supervisor/api/network.py

```diff
@@ -212,25 +232,31 @@ async def interface_update(self, request: web.Request) -> None:
         for key, config in body.items():
             if key == ATTR_IPV4:
                 interface.ipv4setting = IpSetting(
-                    config.get(ATTR_METHOD, InterfaceMethod.STATIC),
-                    config.get(ATTR_ADDRESS, []),
-                    config.get(ATTR_GATEWAY),
-                    config.get(ATTR_NAMESERVERS, []),
+                    method=config.get(ATTR_METHOD, InterfaceMethod.STATIC),
+                    address=config.get(ATTR_ADDRESS, []),
+                    gateway=config.get(ATTR_GATEWAY),
+                    nameservers=config.get(ATTR_NAMESERVERS, []),
                 )
             elif key == ATTR_IPV6:
-                interface.ipv6setting = IpSetting(
-                    config.get(ATTR_METHOD, InterfaceMethod.STATIC),
-                    config.get(ATTR_ADDRESS, []),
-                    config.get(ATTR_GATEWAY),
-                    config.get(ATTR_NAMESERVERS, []),
+                interface.ipv6setting = Ip6Setting(
+                    method=config.get(ATTR_METHOD, InterfaceMethod.STATIC),
+                    addr_gen_mode=config.get(
+                        ATTR_ADDR_GEN_MODE, InterfaceAddrGenMode.DEFAULT_OR_EUI64
```

Comment:
Same here:

```suggestion
                        ATTR_ADDR_GEN_MODE, InterfaceAddrGenMode.DEFAULT
```

I am not sure if this is really relevant in practice as I think the real D-Bus API will always return the default if `addr-gen-mode` never got set. But this makes test work as we expect things.